aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -737,7 +741,11 @@ Tensor& addmm_cpu_out(const Tensor& self, const Tensor& mat1, const Tensor& mat2
   Tensor b_self = std::get<0>(expand_size(self, {mat1.sizes()[0], mat2.sizes()[1]}, "addmm_out"));
   {
     at::NoNamesGuard guard;
-    addmm_impl_cpu_(result, b_self, mat1, mat2, beta, alpha);
```

Comment:
I have concerns with this patch on the aspect that how mkldnn functionality is integrated into PyTorch here.
- first of all, it would be much cleaner to provide a blas-like APIs on BFloat16 (from mkldnn), with LDA, LDB, LDC, etc.
- even we want to hook funcs directly from 'LinearAlgebra.cpp' (which i don't recommend), we can do it from underlying impls e.g. `addmm_impl_cpu_` and `bmm_out_or_baddbmm_` since they have more CHECKs and we can't skip them.
- some internal impls depend on `at::native::cpublas::gemm`, this won't be covered in this patch; again, cpublas has logic to check overlapping memory, we can't pass this.
- `nn.Linear` definition is a little more than direct gemm since it got a bias (need to be broadcasted on output channels), so it is a little different from cblas_sgemm, consider handle this in mkldnn_linear.